src/main/java/lotto/domain/Lotto.java

```diff
@@ -0,0 +1,86 @@
+package lotto.domain;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class Lotto {
+
+    private static final int LOTTO_NUMBER_COUNT = 6;
+
+    private final List<LottoNumber> numbers;
+
+    public Lotto(int... intNumbers) {
+        this(createLottoNumbers(numbersToList(intNumbers)));
+    }
+
+    private static List<Integer> numbersToList(int[] intNumbers) {
+        List<Integer> numbers = new ArrayList<>();
+        for (int number : intNumbers) {
+            numbers.add(number);
+        }
+        return numbers;
+    }
+
+    public Lotto(List<LottoNumber> numbers) {
+        validateSize(numbers);
+        validateDuplicate(numbers);
+        this.numbers = sortNumbers(numbers);
+    }
+
+    public static Lotto from(List<Integer> intNumbers) {
+        return new Lotto(createLottoNumbers(intNumbers));
+    }
+
+    private void validateSize(List<LottoNumber> numbers) {
+        if (numbers.size() != LOTTO_NUMBER_COUNT) {
+            throw new IllegalArgumentException("로또 번호는 총 6개여야 합니다.");
+        }
+    }
+
+    private void validateDuplicate(List<LottoNumber> numbers) {
+        Set<LottoNumber> uniqueNumbers = new HashSet<>(numbers);
+        if (uniqueNumbers.size() != numbers.size()) {
+            throw new IllegalArgumentException("로또 번호는 중복될 수 없습니다.");
+        }
+    }
+
+    private static List<LottoNumber> createLottoNumbers(List<Integer> numbers) {
+        List<LottoNumber> lottoNumbers = new ArrayList<>();
+        for (Integer number : numbers) {
+            lottoNumbers.add(new LottoNumber(number));
+        }
+        return lottoNumbers;
+    }
+
+    private List<LottoNumber> sortNumbers(List<LottoNumber> numbers) {
+        List<LottoNumber> sorted = new ArrayList<>(numbers);
+        sorted.sort(Comparator.comparingInt(LottoNumber::getValue));
+        return sorted;
+    }
+
+    public List<Integer> getNumbers() {
+        List<Integer> result = new ArrayList<>();
+        for (LottoNumber number : numbers) {
+            result.add(number.getValue());
+        }
+        return result;
+    }
+
+    public int countMatchNumber(Lotto other) {
```

Comment:
```suggestion
    public Rank match(Lotto other) {
```
이미 Rank도 enum으로 구현한 상태라면 Lotto가 몇 등인지 여부를 판단하는 것은 어떨까?

Reply:
로또에게 다른로또를 가지고 굳이 숫자가 몇개 맞는지를 확인할 필요없이 몇등인지 묻는방식이 좋은것 같네요